includes/Modules/Idea_Hub.php

```diff
@@ -128,6 +157,96 @@ function( $maybe_empty, $postarr ) {
 		$this->post_topic_setting->register();
 	}
 
+	/**
+	 * Shows admin notification for idea hub ideas on post list screen.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $notices Array of admin notices.
+	 * @return array Array of admin notices.
+	 */
+	private function admin_notice_idea_hub_ideas( $notices ) {
```

Comment:
I think the logic in this method needs to be reworked a little bit to leverage our Notice API. We need to do it like this:

```php
private function admin_notice_idea_hub_ideas( $notices ) {
    $notices[] = new Notice(
        'saved-ideas',
        array(
            'type'            => Notice::TYPE_INFO,
            'dismissible' => true,
            'content'       => function() {
                ob_start();
                // render notice for saved ideas
                return ob_get_clean();
            },
            'active_callback' => function() {
                global $post_type;
                
                $current_screen = get_current_screen();
                if ( is_null( $current_screen ) || 'edit-post' !== $current_screen->id || 'post' !== $post_type ) {
                    return false;
                }

                // pull saved ideas and return true if we have new saved ideas, otherwise, return false
            }
        )
    );


    $notices[] = new Notice(
        'new-ideas',
        array(
            // simmilar approach but for new ideas
        )
    );

    return $notices;
}
```

Reply:
I think we're talking about different ways of implementing the same thing. Our end result is at most one new notice for saved/new ideas using the Notice API

Reply:
Ah... yes, you are right. Please, disregard this comment.

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3668/commits/faadd204be65bbda730317644d3b527de07d8626